src/foundry/prosemirror/image-plugin.d.mts

```diff
@@ -2,15 +2,26 @@ import type { Schema, Slice } from "prosemirror-model";
 import type { Plugin } from "prosemirror-state";
 import type { EditorView } from "prosemirror-view";
 import type ProseMirrorPlugin from "./plugin.d.mts";
+import type { ClientDocumentMixin } from "../client/data/abstract/client-document.d.mts";
 
 export default ProseMirrorImagePlugin;
 /**
  * A class responsible for handle drag-and-drop and pasting of image content. Ensuring no base64 data is injected
  * directly into the journal content and it is instead uploaded to the user's data directory.
  */
 declare class ProseMirrorImagePlugin extends ProseMirrorPlugin {
-  /** {@inheritdoc} */
-  static build(schema: Schema, options?: Record<string, never>): Plugin;
+  /**
+   * @param schema   - The ProseMirror schema.
+   * @param document - A related Document to store extract base64 images for.
+   */
+  constructor(schema: Schema, { document }?: { document?: ClientDocumentMixin<foundry.abstract.Document<any, any>> });
```

Comment:
Ugh I didn't realize some of this was going to be dependent on the new Document implementation. Might be a reason to rebase onto the v11/data-models branch. At a minimum, rewrite the parameter to be `options?: { document: ... }` and then move the document type annotation inside that object. Furthermore, fix the annotation above to match. The actual document typing can be a TODO.